File: Data-Structures-And-Libraries/Python/URI-1022.py
from fractions import gcd

def value(N1, D1, N2, D2: int, operation: chr) -> int:
    if operation == "+":
        first_term = N1 * D2 + N2 * D1
        second_term = D1 * D2

    elif operation == "-":
        first_term = N1 * D2 - N2 * D1
        second_term = D1 * D2

    elif operation == "*":
        first_term = N1 * N2
        second_term = D1 * D2

    else:
        first_term = N1 * D2
        second_term = N2 * D1

    return first_term, second_term


def main():
    while True:
        try:
            cases = int(input())

            for i in range(cases):
                elements = [element for element in input().split()]
                N1, D1, N2, D2 = (
                    int(elements[0]),
                    int(elements[2]),
                    int(elements[4]),
                    int(elements[6]),
                )
                operation = elements[3]

                numbers = value(N1, D1, N2, D2, operation)

                common_divisor = gcd(numbers[0], numbers[1])

                if common_divisor != 1:
                    simplified_1 = int(numbers[0] / common_divisor)
                    simplified_2 = int(numbers[1] / common_divisor)

                    print(
                        str(numbers[0])
                        + "/"
                        + str(numbers[1])
                        + " = "
                        + str(simplified_1)
                        + "/"
                        + str(simplified_2)
                    )

                else:
                    print(
                        str(numbers[0])
                        + "/"
                        + str(numbers[1])
                        + " = "
                        + str(numbers[0])
                        + "/"
                        + str(numbers[1])
                    )

        except EOFError:
            break

main()

File: Data-Structures-And-Libraries/Java/URI-1022.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.math.BigInteger;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.io.InputStreamReader;

public class Main {

	public static void main(String[] args) throws IOException{
		InputStreamReader ir = new InputStreamReader(System.in);
        BufferedReader br = new BufferedReader(ir);
       
        int cases = Integer.parseInt(br.readLine());
        
        int N1, D1, N2, D2;
        
        for(int i = 0; i < cases; i++) {
            String[] elements = br.readLine().split(" ");
            N1 = Integer.valueOf(elements[0]);
            D1 = Integer.valueOf(elements[2]);
            N2 = Integer.valueOf(elements[4]);
            D2 = Integer.valueOf(elements[6]);
            
            String operation = elements[3];
            
            int[] numbers = fractionValue(N1, D1, N2, D2, operation);
            
            BigInteger b1 = BigInteger.valueOf(numbers[0]);
            BigInteger b2 = BigInteger.valueOf(numbers[1]);
            BigInteger gcd = b1.gcd(b2);
            
            int commonDivisor = gcd.intValue();
            
            if(commonDivisor != 1) {
                    int simplified1 = (int) numbers[0] / commonDivisor;
                    int simplified2 = (int) numbers[1] / commonDivisor;
                    
                    System.out.println(String.format("%d/%d = %d/%d", numbers[0], numbers[1],
                                    simplified1, simplified2));
                    
            }
            
            else {
                    System.out.println(String.format("%d/%d = %d/%d", numbers[0], numbers[1],
                                    numbers[0], numbers[1]));
                    
            }
            
        }
    }
           	
	
	static int[] fractionValue(int N1, int D1, int N2, int D2, String operation) {
		
		int firstTerm, secondTerm;
		int[] terms = new int[2];
		
		if(operation.equals("+")){
			
			//System.out.println("Addition");
			firstTerm = N1 * D2 + N2 * D1;
			secondTerm = D1 * D2;
		}
		
		else if(operation.equals("-")){
			
			//System.out.println("Subtration");
			firstTerm = N1 * D2 - N2 * D1;
			secondTerm = D1 * D2;
			
		}
		
		else if(operation.equals("*")) {			
			//System.out.println("Multiplication");
			firstTerm = N1 * N2;
			secondTerm = D1 * D2;
			
		}
		
		else {
			//System.out.println("Division");
			firstTerm = N1 * D2;
			secondTerm = N2 * D1;
			
		}
		
		terms[0] = firstTerm;
		terms[1] = secondTerm;
		
		return terms;
		
	} 
	
}
